src/main/java/noppes/npcs/scripted/ScriptWorld.java

```diff
@@ -346,6 +420,40 @@ public ScriptEntity spawnClone(int x, int y, int z, int tab, String name){
 		return ScriptController.Instance.getScriptForEntity(entity);
 	}
 	
+	public boolean spawnEntity(Entity e) {
+		if (world.spawnEntityInWorld(e)) return true;
+		return false;
+	}
+	
+	/**
+	 * Create a new default faction
+	 * @param name The name of the new faction. If the name already exists an '_' will be added at the end
+	 * @return The faction object which was created
+	 */
+    public ScriptFaction createFaction(String name) {
+    	//System.out.println("Attempting to create new faction "+name);
+    	int maxValue = 0xFFFFFF;
+    	Random r = new Random();
+    	int randomColor = r.nextInt(maxValue + 1);
+    	
+    	Faction fac = new Faction(-1, name, randomColor, 1000);
+    	FactionController.getInstance().saveFaction(fac);
+		
+		return new ScriptFaction(fac);
+    }
+    
+    public ScriptFaction getFactionFromName(String name) {
+    	Faction f = FactionController.getInstance().getFactionFromName(name);
+    	if (f == null) return null;
+    	return new ScriptFaction(f);
+    }
+    
+    public ScriptFaction getFactionFromId(int id) {
+    	Faction f = FactionController.getInstance().get(id);
+    	if (f == null) return null;
+    	return new ScriptFaction(f);
+    }
+	
```

Comment:
Already handled by API.getFactions().

Reply:
Most of my changes to the faction api were just using the FactionController but it makes more sense to just give access to the whole thing. I'll remove these.